haystack/retriever/dense.py

```diff
@@ -62,47 +65,28 @@ def __init__(self,
         """
 
         self.document_store = document_store
-        self.embedding_model = embedding_model
+        self.question_embedding_model = question_embedding_model
+        self.passage_embedding_model = passage_embedding_model
         self.batch_size = batch_size
 
-        #TODO Proper Download + Caching of model if not locally available
-        if embedding_model == "dpr-bert-base-nq":
-            if not Path("models/dpr/checkpoint/retriever/single/nq/bert-base-encoder.cp").is_file():
-                download_dpr(resource_key="checkpoint.retriever.single.nq.bert-base-encoder", out_dir="models/dpr")
-            self.embedding_model = "models/dpr/checkpoint/retriever/single/nq/bert-base-encoder.cp"
-
         if use_gpu and torch.cuda.is_available():
             self.device = torch.device("cuda")
         else:
             self.device = torch.device("cpu")
 
         self.use_amp = use_amp
         self.do_lower_case = do_lower_case
-
-        # Load checkpoint (incl. additional model params)
-        saved_state = load_states_from_checkpoint(self.embedding_model)
-        logger.info('Loaded encoder params:  %s', saved_state.encoder_params)
-        self.do_lower_case = saved_state.encoder_params["do_lower_case"]
-        self.pretrained_model_cfg = saved_state.encoder_params["pretrained_model_cfg"]
-        self.encoder_model_type = saved_state.encoder_params["encoder_model_type"]
-        self.pretrained_file = saved_state.encoder_params["pretrained_file"]
-        self.projection_dim = saved_state.encoder_params["projection_dim"]
-        self.sequence_length = saved_state.encoder_params["sequence_length"]
+        self.sequence_length = sequence_length
+        self.projection_dim = projection_dim
 
         # Init & Load Encoders
-        self.query_encoder = HFBertEncoder.init_encoder(self.pretrained_model_cfg,
-                                                        projection_dim=self.projection_dim,
-                                                        dropout=0.0)
-        self.passage_encoder = HFBertEncoder.init_encoder(self.pretrained_model_cfg,
-                                                          projection_dim=self.projection_dim,
-                                                          dropout=0.0)
-        self.passage_encoder = self._prepare_model(self.passage_encoder, saved_state, prefix="ctx_model.")
-        self.query_encoder = self._prepare_model(self.query_encoder, saved_state, prefix="question_model.")
-        #self.encoder = BiEncoder(question_encoder, ctx_encoder, fix_ctx_encoder=self.fix_ctx_encoder)
-
-        # Load Tokenizer & Tensorizer
-        tokenizer = BertTokenizer.from_pretrained(self.pretrained_model_cfg, do_lower_case=self.do_lower_case)
-        self.tensorizer = BertTensorizer(tokenizer, self.sequence_length)
+        self.question_config = DPRConfig(projection_dim=self.projection_dim, config=self.question_embedding_model, dropout=0.0)
```

Comment:
Please make the naming more consistent. We now have both "question" and "query". We should decide for one and rename all other occurrences. 
I would personally vote for "query" instead of question as DPR could also be used for regular "search queries" when it's trained on such a dataset. What do you think @Timoeller ?

Reply:
I'll change the variable names to `query` and `queries` for consistency